R/qsys_ssh.r

```diff
@@ -55,7 +55,15 @@ SSH = R6::R6Class("SSH",
         fill_options = function(...) {
             args = list(...)
             args$local_port = sub(".*:", "", private$addr)
-            args$ssh.hpc_fwd_port = getOption("clustermq.ssh.hpc_fwd_port", sample(50000:55000, 1))
+            args$ssh.hpc_fwd_port 
+            hpc_fwd_port = getOption(
+                "clustermq.ssh.hpc_fwd_port", 
+                sample((50000:55000),1)
+            )
+            args$ssh.hpc_fwd_port=ifelse(length(hpc_fwd_port)==1, 
+                                            hpc_fwd_port, 
+                                            sample(hpc_fwd_port,1)
+                                        )
```

Comment:
Why not just:

```r
args$ssh.hpc_fwd_port = sample(getOption("clustermq.ssh.hpc_fwd_port", 50000:55000), 1)
```

Reply:
That would be great but if we really want to allow both single ports and port ranges, then your approach would lead to

Reply:
Oh, you're right. I forgot that `sample` with a single integer samples any integer up to and including this point. In that case I'd prefer:

```r
args$ssh.hpc_fwd_port = getOption("clustermq.ssh.hpc_fwd_port", 50000:55000)
if (length(args$ssh.hpc_fwd_port) > 1)
    args$ssh.hpc_fwd_port = sample(args$ssh.hpc_fwd_port, 1)
```

Reply:
That's much cleaner than my algorithm - thanks for this ! I have added this as a commit now.